default/python/AI/colonization/calculate_stability.py

```diff
@@ -158,15 +158,17 @@ def _evaluate_xenophobia(planet, species) -> float:
     if AIDependencies.Tags.XENOPHOBIC not in species.tags:
         return 0.0
     colonies = get_colonized_planets()
-    relevant_systems = within_n_jumps(planet.systemID, 5) & set(colonies.keys())
-    result = 0.0
     universe = fo.getUniverse()
-    value = get_named_real("XENOPHOBIC_SELF_TARGET_HAPPINESS_COUNT")
+    max_jumps = get_named_int("XENOPHOBIC_MAX_JUMPS")
+    relevant_systems = within_n_jumps(planet.systemID, max_jumps)
+    penalty_perjump = get_named_real("XENOPHOBIC_TARGET_STABILITY_PERJUMP")
+    result = 0.0
     for sys_id in relevant_systems:
         for pid in colonies[sys_id]:
```

Comment:
```suggestion
        system = universe.getSystem(sys_id)
        for pid in system.planetIDs:
```
Even if you ignore racial purity, you have to change this, since you want all planets.
Actually your code would crash, since colonies will not have all keys. colonies can actually be removed from the function.

Hopefully we can ignore visibility here and universe.getPlanet(pid).speciesName will simply return "", if the AI doesn't see the planet. But I am not sure what happens, if it had seen it in the past (e.g. due to a derelict scout). Time and test games will tell...